lib/lltz_ir/dsl.ml

```diff
@@ -0,0 +1,184 @@
+let int ?(range = dummy) n = create ~range (LLTZ.E.Const(Int (Z.of_int n)))
+let mutez ?(range = dummy) n = create ~range (LLTZ.E.Const(Mutez (Z.of_int n)))
+let string ?(range = dummy) s = create ~range (LLTZ.E.Const(String s))
+let key ?(range = dummy) s = create ~range (LLTZ.E.Const(Key s))
+let key_hash ?(range = dummy) s = create ~range (LLTZ.E.Const(Key_hash s))
+let bytes ?(range = dummy) s = create ~range (LLTZ.E.Const(Bytes s))
+let chain_id ?(range = dummy) s = create ~range (LLTZ.E.Const(Chain_id s))
+let address_const ?(range = dummy) s = create ~range (LLTZ.E.Const(Address s))
+let timestamp ?(range = dummy) s = create ~range (LLTZ.E.Const(Timestamp s))
+let bls12_381_g1 ?(range = dummy) s = create ~range (LLTZ.E.Const(Bls12_381_g1 s))
+let bls12_381_g2 ?(range = dummy) s = create ~range (LLTZ.E.Const(Bls12_381_g2 s))
+let bls12_381_fr ?(range = dummy) s = create ~range (LLTZ.E.Const(Bls12_381_fr s))
+let signature ?(range = dummy) s = create ~range (LLTZ.E.Const(Signature s))
+
+(* Variables *)
+let var name = LLTZ.E.Var name
+let mut_var name = LLTZ.E.Mut_var name
+
+(* Types *)
+let unit_ty ?(range = dummy) () = LLTZ.T.{ desc = Unit; range }
+let bool_ty ?(range = dummy) () = LLTZ.T.{ desc = Bool; range }
+let nat_ty ?(range = dummy) () = LLTZ.T.{ desc = Nat; range }
+let int_ty ?(range = dummy) () = LLTZ.T.{ desc = Int; range }
+let mutez_ty ?(range = dummy) () = LLTZ.T.{ desc = Mutez; range }
+let string_ty ?(range = dummy) () = LLTZ.T.{ desc = String; range }
+let bytes_ty ?(range = dummy) () = LLTZ.T.{ desc = Bytes; range }
+let chain_id_ty ?(range = dummy) () = LLTZ.T.{ desc = Chain_id; range }
+let timestamp_ty ?(range = dummy) () = LLTZ.T.{ desc = Timestamp; range }
+let address_ty ?(range = dummy) () = LLTZ.T.{ desc = Address; range }
+let key_ty ?(range = dummy) () = LLTZ.T.{ desc = Keys; range }
+let key_hash_ty ?(range = dummy) () = LLTZ.T.{ desc = Key_hash; range }
+let signature_ty ?(range = dummy) () = LLTZ.T.{ desc = Signature; range }
+
+(* Expressions *)
+let variable ?(range = dummy) var = create ~range (LLTZ.E.Variable var)
+let let_in ?(range = dummy) (var, rhs, in_) = create ~range (LLTZ.E.Let_in { let_var = var; rhs; in_ })
+let lambda ?(range = dummy) (var, return_type, body) = create ~range (LLTZ.E.Lambda { lam_var = var; return_type; body })
+let lambda_rec ?(range = dummy) (var, mu, return_type, body) = create ~range (LLTZ.E.Lambda_rec { lam_var = var; mu_var = mu; return_type; body })
+let app ?(range = dummy) (abs, arg) = create ~range (LLTZ.E.App { abs; arg })
+let let_mut_in ?(range = dummy) (var, rhs, in_) = create ~range (LLTZ.E.Let_mut_in { let_var = var; rhs; in_ })
+let deref ?(range = dummy) var = create ~range (LLTZ.E.Deref var)
+let assign ?(range = dummy) (var, value) = create ~range (LLTZ.E.Assign (var, value))
+let if_bool ?(range = dummy) (condition, then_, else_) = create ~range (LLTZ.E.If_bool { condition; if_true = then_; if_false = else_ })
```

Comment:
I was wondering if it wouldn't be better to ask the caller to name the parts of the argument (with a record instead of a tuple), and define here:
````
let if_bool ?(range=dummy) arguments = create ~range (LLTZ.E.If_bool arguments)
```

Reply:
It seems to me that the caller would then have to remember what exact names to use, which complicates the usage of the dsl. There will be another PR that actually tests the dsl and its compilation, so I guess its better to merge this and then we can check there if it looks good enough with just the tuples or would be more readable with added named args. what do you think?

Reply:
actually realised that the shape of the record is determined by the expression (If_bool) in this case. still I would rather make this modification later once rewriting the tests as it is mainly related to readbility/usability, so its better to see how it actually looks like.

Reply:
AS long as you don't forget to do the assessment later, it's fine. :)